tests/v2/validation/pipeline/airgapsetup/Jenkinsfile.airgap.setup.delete

```diff
@@ -0,0 +1,118 @@
+              deleteDir()
+              checkout([
+                        $class: 'GitSCM',
+                        branches: [[name: "*/${branch}"]],
+                        extensions: scm.extensions + [[$class: 'CleanCheckout']],
+                        userRemoteConfigs: repo
+                      ])
+            }
+        stage('Configure and Build') {
+          sh "docker build . -f Dockerfile --build-arg TERRAFORM_VERSION=${env.TERRAFORM_VERSION} -t ${imageName}"
+        }
+        stage('Setup'){
+            sh returnStdout: true, script: 'wget -qO ./yq https://github.com/mikefarah/yq/releases/latest/download/yq_linux_amd64'
+            sh returnStdout:true, script: 'chmod a+x ./yq'
+
+            println "AWS s3 config ${s3Config}"
+            
+            AWS_ACCESS_KEY_ID = sh (
+                  script: "echo '${s3Config}' | ./yq '.accessKey'",
+                  returnStdout: true
+                ).trim()
+            println "AWS_ACCESS_KEY_ID: ${AWS_ACCESS_KEY_ID}"
+
+            AWS_SECRET_ACCESS_KEY = sh (
+                  script: "echo '${s3Config}' | ./yq '.secretKey'",
+                  returnStdout: true
+                ).trim()
+            
+            println "AWS_SECRET_ACCESS_KEY: ${AWS_SECRET_ACCESS_KEY}"
+            
+            AWS_S3_BUCKET = sh (
+                  script: "echo '${s3Config}' | ./yq '.s3Bucket'",
+                  returnStdout: true
+                ).trim()
+                
+            println "AWS_S3_BUCKET: ${AWS_S3_BUCKET}"
+
+            AWS_S3_REGION =    sh (
+                  script: "echo '${s3Config}' | ./yq '.s3BucketRegion'",
+                  returnStdout: true
+                ).trim()
+            println "AWS_S3_REGION: ${AWS_S3_REGION}"
+
+        }
+
+        stage('Cleanup setup'){
+            script {
+            sh """
+                docker volume create $terraformVolume
+
+                docker run --name $s3Container \
+                -e AWS_ACCESS_KEY_ID=$AWS_ACCESS_KEY_ID \
+                -e AWS_SECRET_ACCESS_KEY=$AWS_SECRET_ACCESS_KEY \
+                -e AWS_DEFAULT_REGION=$AWS_S3_REGION \
+                -v $terraformVolume:/terraform-files  \
+                amazon/aws-cli \
+                s3 cp s3://$AWS_S3_BUCKET/terraform/ /terraform-files --recursive
+
+                docker run --rm -t -v $terraformVolume:/terraform-files --name $testContainer ${imageName} \
+                sh -c "chmod +x -R /terraform-files && terraform -chdir=/terraform-files init && terraform -chdir=/terraform-files destroy -auto-approve"
+
+                docker run --rm \
+                -e AWS_ACCESS_KEY_ID=$AWS_ACCESS_KEY_ID \
+                -e AWS_SECRET_ACCESS_KEY=$AWS_SECRET_ACCESS_KEY \
+                -e AWS_DEFAULT_REGION=$AWS_S3_REGION \
+                amazon/aws-cli \
+                s3 rm s3://$AWS_S3_BUCKET/terraform/ --recursive
+
+                docker rm $s3Container
+                docker volume rm $terraformVolume
```

Comment:
can you add a try / catch here and add this cleanup to the catch as well just incase the script fails